langkit/dsl_unparse.py

```diff
@@ -118,25 +285,45 @@ def is_simple_expr(expr):
 
 def emit_indent_expr(expr, **ctx):
     strn = emit_expr(expr, **ctx)
-    if len(strn) > 40:
+    starts_with_comment = len(strn) > 0 and strn[0] == '#'
+    if len(strn) > 40 or starts_with_comment:
         return "$i$hl{}$d$hl".format(strn)
     else:
         return strn
 
 
+def prepend_comments(expr, **ctx):
+    arg_expr = ctx.pop('arg_expr', None)
+    if arg_expr is None:
+        return emit_expr(expr, **ctx), False
+
+    walker = ctx.get('walker')
+    with walker.arg(arg_expr):
+        coms = walker.emit_comments()
+        strn = emit_expr(expr, **ctx)
+
+    if coms == "":
+        return strn, False
+
+    return coms + strn, True
+
+
 def emit_paren_expr(expr, **ctx):
     from langkit.expressions import Let
-    if is_simple_expr(expr):
-        strn = emit_expr(expr, **ctx)
-        return emit_paren(strn) if len(strn) > 40 else strn
-    elif isinstance(expr, Let):
+
+    if isinstance(expr, Let):
         return emit_expr(expr, **ctx)
+
+    strn, has_coms = prepend_comments(expr, **ctx)
```

Comment:
Not bloçcking but it would be good to understand why the call to `prepend_comments` is made inside `emit_paren_expr`